### ISSUE TITLE
Create CLI command hierarchy

### ISSUE THREAD

**Issue by CodeSandwich** [radicle-dev/radicle-registry#268]
### Problem
We have too many long, specific CLI commands. E.g. for `-org` we have `list-orgs`, `show-org`, `register-org` and `unregister-org`, it's the same with `-user` and similar with `-project`. The list is growing quickly and so does the user confusion.

### Solution
As a user I would prefer having subcommands, e.g. `org list`, `org show`, `org register` and `org unregister`. This way it would be easier for me to discover commands by having a list of general topics of interest in a top-level `--help` and a list of topic-specific actions in `org --help`.

**Comment by NunoAlexandre** [comment]
@CodeSandwich thanks for starting this discussion. As part of the [Wallet CLI](#275) work, this was something I started to look into.

I have a few alternatives I'd like us to consider, where yours is one of them :)

### Commands

First, let's enumerate the commands we have atm:

``` rust
enum Command {
    ListOrgs(ListOrgs),
    ListProjects(ListProjects),
    RegisterOrg(RegisterOrg),
    UnregisterOrg(UnregisterOrg),
    RegisterProject(RegisterProject),
    RegisterUser(RegisterUser),
    UnregisterUser(UnregisterUser),
    ShowBalance(ShowBalance),
    ShowGenesisHash(ShowGenesisHash),
    ShowOrg(ShowOrg),
    ShowProject(ShowProject),
    Transfer(Transfer),
    TransferOrgFunds(TransferOrgFunds),
    ShowAddress(ShowAddress),
}
```

## Alternative A - Organized by domains

Organize the commands by their domain. `Org`-related commands go under `org`, `Project`-related commands go under `project`, and so on.

Following this alternative, the commands would look like this:

``` bash
> cli org list
> cli org register
> cli org unregister
> cli org show
> cli org transfer

> cli project list
> cli project register
> cli project show

> cli user register
> cli user unregister

> cli account address
> cli account transfer
> cli account balance

> cli genesis show
> 
```

### Pros

+ Natural categorization
+ Very readable


## Alternative B - CQRS-inspired

Some of the _commands_ translate to transactions, others are just _queries_.
 

``` bash

> cli submit org-register
> cli submit org-unregister
> cli submit org transfer

> cli query org-list
> cli query org-show

> cli query project-list
> cli query project-show

> cli submit project-register

> cli submit user register
> cli submit user unregister

> cli query address
> cli query balance
> cli submit transfer

> cli query genesis-show
```

### Pros

+ Separates the commands by their nature and effect in the system
+ Hey, CQRS is _the_ answer, right?

### Cons

- Basically just splits the current problem into two sections, `submit` and `query`.

💬 When I first thought of this alternative, my idea was to have `cli submit`, `cli query`, `cli wallet`. etc. I.e, each subsection _could_ be an executable of its own.


@geigerzaehler @geigerzaehler thoughts?

**Comment by geigerzaehler** [comment]
I’d prefer option A.

**Comment by CodeSandwich** [comment]
Option A seems to be a much better fit for a CLI.

What do you think about creating a `dev` domain? We could put there all the unstable, poorly documented and dangerous tools so useful for development, but not intended for general public. This way they could be easily isolated and toggled with a feature.

**Comment by NunoAlexandre** [comment]
yeah I also like option A better.

> What do you think about creating a dev domain? We could put there all the unstable, poorly documented and dangerous tools so useful for development, but not intended for general public. This way they could be easily isolated and toggled with a feature.

I don't think we need that yet, but it's an idea. How would you feature-flag that though?

**Comment by CodeSandwich** [comment]
Ususally CLI subcommands are implemented in `StructOpt` with enums, we can `#[cfg(feature(dev_cli))]` the `Dev` variant, it's `match`es and the `mod` containing the `dev` commands implementations.
